chrome/content/zotero/xpcom/data/relations.js

```diff
@@ -156,13 +156,16 @@ Zotero.Relations = new function () {
 		}
 		var toReturn = [];
 		for (let predicateID in o) {
-			o[predicateID].forEach(subjectID => toReturn.push({
-				subject: objectsClass.get(subjectID),
-				predicate: Zotero.RelationPredicates.getName(predicateID)
+			o[predicateID].forEach(Zotero.Promise.coroutine(function* (subjectID) {
+				toReturn.push({
+					subject: yield objectsClass.getAsync(subjectID),
+					predicate: Zotero.RelationPredicates.getName(predicateID)
+				})
+				return toReturn;
```

Comment:
I think the line above is unnecessary. The array accumulates as a side-effect of the function and is returned after the loop completes. The inner return value isn't used for anything.

Reply:
Oops, this breaks tests. Will replace with a *much* simpler change a bit later, plus a test of the conditions that originally triggered the failure in my environment.